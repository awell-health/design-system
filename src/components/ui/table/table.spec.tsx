import { render } from '@testing-library/react';
import { expect, it, describe } from 'vitest';

import {
  Table,
  TableHeader,
  TableCaption,
  TableBody,
  TableHead,
  TableRow,
  TableCell
} from './index';

describe('Table', () => {
  it('match snapshot', () => {
    const result = render(
      <Table>
        <TableHeader>
          <TableRow>
            <TableHead info={<span>Info</span>}>Carflow name</TableHead>
            <TableHead sortable={true}>Version</TableHead>
          </TableRow>
        </TableHeader>
        <TableBody>
          {[...Array(10).keys()].map((i) => (
            <TableRow key={i}>
              <TableCell>Care Flow Name-{i}</TableCell>
              <TableCell supportingText='by example@awellhealth.com'>Example Cell</TableCell>
            </TableRow>
          ))}
        </TableBody>
        <TableCaption className='py-0'>Table Caption</TableCaption>
      </Table>
    );

    expect(result).toMatchSnapshot();
  });
});
