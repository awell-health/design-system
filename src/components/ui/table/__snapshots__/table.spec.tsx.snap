// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Table Snaphot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="relative w-full rounded-lg border border-slate-200 flex-col"
      >
        <table
          class="w-full caption-bottom text-sm"
        >
          <thead
            class="[&_tr]:border-b [&_tr:first-child>th:first-child]:rounded-tl-lg [&_tr:first-child>th:last-child]:rounded-tr-lg"
          >
            <tr
              class="border-b border-slate-100 hover:bg-slate-50 [&_td:last-child]:text-right"
            >
              <th
                class="px-6 py-3 bg-gray-50 text-slate-500 text-sm font-medium leading-[18px] text-left group hover:text-slate-700"
              >
                <span
                  class="flex gap-1"
                >
                  Carflow name
                   
                  <svg
                    class="remixicon fill-gray-500 group-hover:fill-slate-700"
                    fill="default"
                    height="16"
                    viewBox="0 0 24 24"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 22C6.47715 22 2 17.5228 2 12C2 6.47715 6.47715 2 12 2C17.5228 2 22 6.47715 22 12C22 17.5228 17.5228 22 12 22ZM12 20C16.4183 20 20 16.4183 20 12C20 7.58172 16.4183 4 12 4C7.58172 4 4 7.58172 4 12C4 16.4183 7.58172 20 12 20ZM11 15H13V17H11V15ZM13 13.3551V14H11V12.5C11 11.9477 11.4477 11.5 12 11.5C12.8284 11.5 13.5 10.8284 13.5 10C13.5 9.17157 12.8284 8.5 12 8.5C11.2723 8.5 10.6656 9.01823 10.5288 9.70577L8.56731 9.31346C8.88637 7.70919 10.302 6.5 12 6.5C13.933 6.5 15.5 8.067 15.5 10C15.5 11.5855 14.4457 12.9248 13 13.3551Z"
                    />
                  </svg>
                </span>
              </th>
              <th
                class="px-6 py-3 bg-gray-50 text-slate-500 text-sm font-medium leading-[18px] text-left group hover:text-slate-700 cursor-pointer"
              >
                <span
                  class="flex gap-1"
                >
                  Version
                   
                </span>
              </th>
            </tr>
          </thead>
          <tbody
            class="[&_tr:last-child]:border-0 [&_tr:last-child>td:first-child]:rounded-bl-lg [&_tr:last-child>td:last-child]:rounded-br-lg"
          >
            <tr
              class="border-b border-slate-100 hover:bg-slate-50 [&_td:last-child]:text-right"
            >
              <td
                class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
              >
                <span
                  class="flex gap-1 flex-col items-start"
                >
                  Care Flow Name-
                  0
                </span>
              </td>
              <td
                class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
              >
                <span
                  class="flex gap-1 flex-col items-start"
                >
                  Example Cell
                  <span
                    class="text-slate-500 font-normal"
                  >
                    by example@awellhealth.com
                  </span>
                </span>
              </td>
            </tr>
            <tr
              class="border-b border-slate-100 hover:bg-slate-50 [&_td:last-child]:text-right"
            >
              <td
                class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
              >
                <span
                  class="flex gap-1 flex-col items-start"
                >
                  Care Flow Name-
                  1
                </span>
              </td>
              <td
                class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
              >
                <span
                  class="flex gap-1 flex-col items-start"
                >
                  Example Cell
                  <span
                    class="text-slate-500 font-normal"
                  >
                    by example@awellhealth.com
                  </span>
                </span>
              </td>
            </tr>
            <tr
              class="border-b border-slate-100 hover:bg-slate-50 [&_td:last-child]:text-right"
            >
              <td
                class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
              >
                <span
                  class="flex gap-1 flex-col items-start"
                >
                  Care Flow Name-
                  2
                </span>
              </td>
              <td
                class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
              >
                <span
                  class="flex gap-1 flex-col items-start"
                >
                  Example Cell
                  <span
                    class="text-slate-500 font-normal"
                  >
                    by example@awellhealth.com
                  </span>
                </span>
              </td>
            </tr>
            <tr
              class="border-b border-slate-100 hover:bg-slate-50 [&_td:last-child]:text-right"
            >
              <td
                class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
              >
                <span
                  class="flex gap-1 flex-col items-start"
                >
                  Care Flow Name-
                  3
                </span>
              </td>
              <td
                class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
              >
                <span
                  class="flex gap-1 flex-col items-start"
                >
                  Example Cell
                  <span
                    class="text-slate-500 font-normal"
                  >
                    by example@awellhealth.com
                  </span>
                </span>
              </td>
            </tr>
            <tr
              class="border-b border-slate-100 hover:bg-slate-50 [&_td:last-child]:text-right"
            >
              <td
                class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
              >
                <span
                  class="flex gap-1 flex-col items-start"
                >
                  Care Flow Name-
                  4
                </span>
              </td>
              <td
                class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
              >
                <span
                  class="flex gap-1 flex-col items-start"
                >
                  Example Cell
                  <span
                    class="text-slate-500 font-normal"
                  >
                    by example@awellhealth.com
                  </span>
                </span>
              </td>
            </tr>
            <tr
              class="border-b border-slate-100 hover:bg-slate-50 [&_td:last-child]:text-right"
            >
              <td
                class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
              >
                <span
                  class="flex gap-1 flex-col items-start"
                >
                  Care Flow Name-
                  5
                </span>
              </td>
              <td
                class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
              >
                <span
                  class="flex gap-1 flex-col items-start"
                >
                  Example Cell
                  <span
                    class="text-slate-500 font-normal"
                  >
                    by example@awellhealth.com
                  </span>
                </span>
              </td>
            </tr>
            <tr
              class="border-b border-slate-100 hover:bg-slate-50 [&_td:last-child]:text-right"
            >
              <td
                class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
              >
                <span
                  class="flex gap-1 flex-col items-start"
                >
                  Care Flow Name-
                  6
                </span>
              </td>
              <td
                class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
              >
                <span
                  class="flex gap-1 flex-col items-start"
                >
                  Example Cell
                  <span
                    class="text-slate-500 font-normal"
                  >
                    by example@awellhealth.com
                  </span>
                </span>
              </td>
            </tr>
            <tr
              class="border-b border-slate-100 hover:bg-slate-50 [&_td:last-child]:text-right"
            >
              <td
                class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
              >
                <span
                  class="flex gap-1 flex-col items-start"
                >
                  Care Flow Name-
                  7
                </span>
              </td>
              <td
                class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
              >
                <span
                  class="flex gap-1 flex-col items-start"
                >
                  Example Cell
                  <span
                    class="text-slate-500 font-normal"
                  >
                    by example@awellhealth.com
                  </span>
                </span>
              </td>
            </tr>
            <tr
              class="border-b border-slate-100 hover:bg-slate-50 [&_td:last-child]:text-right"
            >
              <td
                class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
              >
                <span
                  class="flex gap-1 flex-col items-start"
                >
                  Care Flow Name-
                  8
                </span>
              </td>
              <td
                class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
              >
                <span
                  class="flex gap-1 flex-col items-start"
                >
                  Example Cell
                  <span
                    class="text-slate-500 font-normal"
                  >
                    by example@awellhealth.com
                  </span>
                </span>
              </td>
            </tr>
            <tr
              class="border-b border-slate-100 hover:bg-slate-50 [&_td:last-child]:text-right"
            >
              <td
                class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
              >
                <span
                  class="flex gap-1 flex-col items-start"
                >
                  Care Flow Name-
                  9
                </span>
              </td>
              <td
                class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
              >
                <span
                  class="flex gap-1 flex-col items-start"
                >
                  Example Cell
                  <span
                    class="text-slate-500 font-normal"
                  >
                    by example@awellhealth.com
                  </span>
                </span>
              </td>
            </tr>
          </tbody>
          <caption
            class="border-t border-slate-200 py-0"
          >
            Table Caption
          </caption>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="relative w-full rounded-lg border border-slate-200 flex-col"
    >
      <table
        class="w-full caption-bottom text-sm"
      >
        <thead
          class="[&_tr]:border-b [&_tr:first-child>th:first-child]:rounded-tl-lg [&_tr:first-child>th:last-child]:rounded-tr-lg"
        >
          <tr
            class="border-b border-slate-100 hover:bg-slate-50 [&_td:last-child]:text-right"
          >
            <th
              class="px-6 py-3 bg-gray-50 text-slate-500 text-sm font-medium leading-[18px] text-left group hover:text-slate-700"
            >
              <span
                class="flex gap-1"
              >
                Carflow name
                 
                <svg
                  class="remixicon fill-gray-500 group-hover:fill-slate-700"
                  fill="default"
                  height="16"
                  viewBox="0 0 24 24"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 22C6.47715 22 2 17.5228 2 12C2 6.47715 6.47715 2 12 2C17.5228 2 22 6.47715 22 12C22 17.5228 17.5228 22 12 22ZM12 20C16.4183 20 20 16.4183 20 12C20 7.58172 16.4183 4 12 4C7.58172 4 4 7.58172 4 12C4 16.4183 7.58172 20 12 20ZM11 15H13V17H11V15ZM13 13.3551V14H11V12.5C11 11.9477 11.4477 11.5 12 11.5C12.8284 11.5 13.5 10.8284 13.5 10C13.5 9.17157 12.8284 8.5 12 8.5C11.2723 8.5 10.6656 9.01823 10.5288 9.70577L8.56731 9.31346C8.88637 7.70919 10.302 6.5 12 6.5C13.933 6.5 15.5 8.067 15.5 10C15.5 11.5855 14.4457 12.9248 13 13.3551Z"
                  />
                </svg>
              </span>
            </th>
            <th
              class="px-6 py-3 bg-gray-50 text-slate-500 text-sm font-medium leading-[18px] text-left group hover:text-slate-700 cursor-pointer"
            >
              <span
                class="flex gap-1"
              >
                Version
                 
              </span>
            </th>
          </tr>
        </thead>
        <tbody
          class="[&_tr:last-child]:border-0 [&_tr:last-child>td:first-child]:rounded-bl-lg [&_tr:last-child>td:last-child]:rounded-br-lg"
        >
          <tr
            class="border-b border-slate-100 hover:bg-slate-50 [&_td:last-child]:text-right"
          >
            <td
              class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
            >
              <span
                class="flex gap-1 flex-col items-start"
              >
                Care Flow Name-
                0
              </span>
            </td>
            <td
              class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
            >
              <span
                class="flex gap-1 flex-col items-start"
              >
                Example Cell
                <span
                  class="text-slate-500 font-normal"
                >
                  by example@awellhealth.com
                </span>
              </span>
            </td>
          </tr>
          <tr
            class="border-b border-slate-100 hover:bg-slate-50 [&_td:last-child]:text-right"
          >
            <td
              class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
            >
              <span
                class="flex gap-1 flex-col items-start"
              >
                Care Flow Name-
                1
              </span>
            </td>
            <td
              class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
            >
              <span
                class="flex gap-1 flex-col items-start"
              >
                Example Cell
                <span
                  class="text-slate-500 font-normal"
                >
                  by example@awellhealth.com
                </span>
              </span>
            </td>
          </tr>
          <tr
            class="border-b border-slate-100 hover:bg-slate-50 [&_td:last-child]:text-right"
          >
            <td
              class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
            >
              <span
                class="flex gap-1 flex-col items-start"
              >
                Care Flow Name-
                2
              </span>
            </td>
            <td
              class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
            >
              <span
                class="flex gap-1 flex-col items-start"
              >
                Example Cell
                <span
                  class="text-slate-500 font-normal"
                >
                  by example@awellhealth.com
                </span>
              </span>
            </td>
          </tr>
          <tr
            class="border-b border-slate-100 hover:bg-slate-50 [&_td:last-child]:text-right"
          >
            <td
              class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
            >
              <span
                class="flex gap-1 flex-col items-start"
              >
                Care Flow Name-
                3
              </span>
            </td>
            <td
              class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
            >
              <span
                class="flex gap-1 flex-col items-start"
              >
                Example Cell
                <span
                  class="text-slate-500 font-normal"
                >
                  by example@awellhealth.com
                </span>
              </span>
            </td>
          </tr>
          <tr
            class="border-b border-slate-100 hover:bg-slate-50 [&_td:last-child]:text-right"
          >
            <td
              class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
            >
              <span
                class="flex gap-1 flex-col items-start"
              >
                Care Flow Name-
                4
              </span>
            </td>
            <td
              class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
            >
              <span
                class="flex gap-1 flex-col items-start"
              >
                Example Cell
                <span
                  class="text-slate-500 font-normal"
                >
                  by example@awellhealth.com
                </span>
              </span>
            </td>
          </tr>
          <tr
            class="border-b border-slate-100 hover:bg-slate-50 [&_td:last-child]:text-right"
          >
            <td
              class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
            >
              <span
                class="flex gap-1 flex-col items-start"
              >
                Care Flow Name-
                5
              </span>
            </td>
            <td
              class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
            >
              <span
                class="flex gap-1 flex-col items-start"
              >
                Example Cell
                <span
                  class="text-slate-500 font-normal"
                >
                  by example@awellhealth.com
                </span>
              </span>
            </td>
          </tr>
          <tr
            class="border-b border-slate-100 hover:bg-slate-50 [&_td:last-child]:text-right"
          >
            <td
              class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
            >
              <span
                class="flex gap-1 flex-col items-start"
              >
                Care Flow Name-
                6
              </span>
            </td>
            <td
              class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
            >
              <span
                class="flex gap-1 flex-col items-start"
              >
                Example Cell
                <span
                  class="text-slate-500 font-normal"
                >
                  by example@awellhealth.com
                </span>
              </span>
            </td>
          </tr>
          <tr
            class="border-b border-slate-100 hover:bg-slate-50 [&_td:last-child]:text-right"
          >
            <td
              class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
            >
              <span
                class="flex gap-1 flex-col items-start"
              >
                Care Flow Name-
                7
              </span>
            </td>
            <td
              class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
            >
              <span
                class="flex gap-1 flex-col items-start"
              >
                Example Cell
                <span
                  class="text-slate-500 font-normal"
                >
                  by example@awellhealth.com
                </span>
              </span>
            </td>
          </tr>
          <tr
            class="border-b border-slate-100 hover:bg-slate-50 [&_td:last-child]:text-right"
          >
            <td
              class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
            >
              <span
                class="flex gap-1 flex-col items-start"
              >
                Care Flow Name-
                8
              </span>
            </td>
            <td
              class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
            >
              <span
                class="flex gap-1 flex-col items-start"
              >
                Example Cell
                <span
                  class="text-slate-500 font-normal"
                >
                  by example@awellhealth.com
                </span>
              </span>
            </td>
          </tr>
          <tr
            class="border-b border-slate-100 hover:bg-slate-50 [&_td:last-child]:text-right"
          >
            <td
              class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
            >
              <span
                class="flex gap-1 flex-col items-start"
              >
                Care Flow Name-
                9
              </span>
            </td>
            <td
              class="px-6 py-4 justify-start items-center flex-inline text-slate-800 text-sm font-medium leading-tight text-left"
            >
              <span
                class="flex gap-1 flex-col items-start"
              >
                Example Cell
                <span
                  class="text-slate-500 font-normal"
                >
                  by example@awellhealth.com
                </span>
              </span>
            </td>
          </tr>
        </tbody>
        <caption
          class="border-t border-slate-200 py-0"
        >
          Table Caption
        </caption>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
